### DISCUSSION TITLE
How can i change submit button text when rendering form (without changing it in admin settings)

### DISCUSSION THREAD

**Discussion by piotrpog** [verbb/formie#2111] [Q&A]
I can change field settings like this, but its for specific field grabbed by handle:
```
{% do form.setFieldSettings(fieldHandle, {
    required: true,
}) %}
```
Submit buttons however do not have handles. Is there maybe option for affecting all fields of specific type? So i would just affect all submit buttons.

**Comment by engram-design** [comment]
This is currently only possible in [Formie 3](https://verbb.io/craft-plugins/formie/docs/template-guides/rendering-pages#override-page-settings) at the moment.

```twig
{% set form = craft.formie.forms.handle('contactForm').one() %}

{% do form.setPageSettings('page-1', {
    {# Override the submit button text for "Page 1" #}
    submitButtonLabel: 'Click here to submit',
}) %}

{{ craft.formie.renderForm(form) }}
```
